### PR TITLE
feat: move less-specific options to lessOptions

### PR DESCRIPTION
**Pull request by ryanclark**
This PR contains a:

- [ ] **bugfix**
- [X] new **feature**
- [X] **code refactor**
- [X] **test update** 
- [ ] **typo fix**
- [ ] **metadata update**

### Motivation / Use-Case

Move less-specific options over to `lessOptions`, in parity with what `sass-loader` does. Add schema validation using JSON schema and `schema-utils` also.

### Breaking Changes

This contains a breaking change, as options passed through to less now need to be under `lessOptions`.

Respects the `devtool` setting from webpack also, if `options.sourceMap` is not set

Remove the `sourceMap.outputSourceFiles` option and just respect `options.sourceMap` as a `boolean`

### CODE REVIEW

**Comment by ryanclark** [comment]
I also refactored the README as one line statements were split across multiple lines, making editing it a pain

File: src/index.js [hunk 1]
```diff
@@ -18,7 +28,30 @@ function lessLoader(source) {
     );
   }
 
-  processResult(loaderContext, render(source, options));
+  const lessOptions = {
```
Comment (evilebottnawi) [inline src/index.js]: Let's move it to `getLessOptions` (example from sass-loader https://github.com/webpack-contrib/sass-loader/blob/master/src/getSassOptions.js)

**Comment by anikethsaha** [comment]
Seems like it similar to https://github.com/webpack-contrib/less-loader/pull/310

can you confirm this ?

File: src/options.json [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+        {
+          "type": "object",
+          "additionalProperties": true
+        },
+        {
+          "instanceof": "Function"
+        }
+      ]
+    },
+    "sourceMap": {
```
Comment (evilebottnawi) [inline src/options.json]: We should allow only `boolean` value,  `outputSourceFiles` it is the less specificity option, we should not respect it

**Pull request merged**